src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -145,6 +167,34 @@ public bool Equals(ArraySegment<T> obj)
             return obj._array == _array && obj._offset == _offset && obj._count == _count;
         }
 
+        public ArraySegment<T> Slice(int index)
```

Comment:
@KrzysztofCwalina Is this blocking this PR? In the case they decide to add a `Range` struct we can just add another overload in addition to these two.